### DISCUSSION TITLE
"tissue_membership" in LSQ method

### DISCUSSION THREAD

**Discussion by Siji-chl** [jcreinhold/intensity-normalization#56] [Q&A] [answered]
Hi @jcreinhold , I m sorry for not replying you in time in issue #55 After reading the original paper, I found that RAVEL method does not suitable for my data.

Now, I turn to LSQ method. However, these is no algorithm descriptions about LSQ in doc, as well as you pre-print paper. So I directly ran the code according to the tutorial. It ran into **line 97** `csf_mean = np.average(image, weights=tissue_membership[..., 0])` with error `TypeError: Axis must be specified when shapes of a and weights differ.`. Also, I m confused that why this methodonly need one image to be fitted this method? Do images also need to be co-registrated like RAVEL? 

Thank you advanced and looking to your reply!
Best, chl

**Comment by jcreinhold** [comment]
I'll add a description of the LSQ method soon.

Can you describe your data? Is it DICOM, NIfTI, etc.? Is it skull-stripped? Or are you providing the brain masks?

The images don't have to be coregistered for LSQ. This looks like a bug. Can you fill out a bug report for the error please? Please provide the version numbers and the code you used to produce the error (changing names of files or strings is fine if there is sensitive info there).

Thanks for the feedback!

**Comment by jcreinhold** [comment]
@Siji-chl Were you able to resolve the issue or did you move on? I'm not able to reproduce the error you're describing. Did you have any other context to provide?

Thanks!

**Comment by Siji-chl** [comment, marked as answer]
@jcreinhold I'm very sorry. I have submitted a BUG report for the error. Thanks a lot!
